leash.go

```diff
@@ -377,28 +377,45 @@ func modifyEventContents(toBeSent chan event.Event, options GlobalOptions) chan
 					for k, v := range parsedAddFields {
 						ev.Data[k] = v
 					}
-					ev.SampleRate = int(options.SampleRate)
-					if dynamicSampler != nil {
-						key := makeDynsampleKey(&ev, options)
-						sr := dynamicSampler.GetSampleRate(key)
-						if rand.Intn(sr) != 0 {
-							ev.SampleRate = -1
-						} else {
-							ev.SampleRate = sr
+					// get presampled field if it exists
+					if options.PreSampledField != "" {
+						var presampledRate int
+						if psr, ok := ev.Data[options.PreSampledField]; ok {
+							switch psr := psr.(type) {
+							case float64:
+								presampledRate = int(psr)
+							case string:
+								if val, err := strconv.Atoi(psr); err == nil {
```

Comment:
are we doing anything with this error?

Reply:
probably should but honeytail doesn't have great support for communicating this sort of thing. :/